### PR TITLE
Github Integrations

### PR DESCRIPTION
**Pull request by Salman2301**
# Description 📣

This PR is focus on updating the current Github Integrations and also add Organization and Environment sync.

- [x] Move the current integrations to `react-hook-form`
  - [x] Show error on bad request
- [x] Organization (UI)
- [x] Environment (UI)
- [x] Integrations Section (UI)
- [x] Sync.ts backend
- [ ] Refactor
- [x] Update docs

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Documentation

# Tests 🛠️
Will update soon

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

**Comment by Salman2301** [comment]
I am keep getting 404 issue `/api/v1/integration-auth/{integrationAuthId}/github/orgs` for this route. I did add the routes and spec in the last commit.
Will look into it later

**Comment by Salman2301** [comment]
integrations section for 
1. Github Repo
1. Github Environment
1. Github Organization 

Note that the schema being used was mongo. I updated the schema to match the current pg table. And also `scope` field moved outside from the metadata. Which `Qovery` integration was being used. Would be nice @vmatsiiako and @dangtony98 if you take a look.

Screenshot 2024-02-15 at 3 55 41 PM

**Comment by Salman2301** [comment]
Based on different scope, the secret will be synced

Screenshot 2024-02-15 at 4 07 46 PM
Screenshot 2024-02-15 at 4 07 54 PM
Screenshot 2024-02-15 at 4 08 18 PM

**Comment by Salman2301** [comment]
@akhilmhdh should I move to zod form resolver instead of yup?

**Comment by Salman2301** [comment]
This is ready to merge

File: docs/integrations/cicd/githubactions.mdx [hunk 1]
```diff
@@ -5,14 +5,26 @@ description: "How to sync secrets from Infisical to GitHub Actions"
...
-    - Set up and add envars to [Infisical Cloud](https://app.infisical.com)
-    - Ensure you have admin privileges to the repo you want to sync secrets to.
+
+    <Info>
+     To know more about usage and limitation of secrets in GitHub Actions, please visit the [Official Github documentation](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions).
+    </Info>
+
+
+    #### Prerequisites:
+      - Set up and add envars to [Infisical Cloud](https://app.infisical.com)
```
[the remark this answers is not in the document]
Comment (Salman2301) [reply docs/integrations/cicd/githubactions.mdx:22]: this line "Set up and add envars" is the same line used for all the integrations for some reason (51 files have the same line). I am inclined to change it too. But would be nice anyone from Infisical team did it.

**Comment by maidul98** [comment]
Sorry about the delay @Salman2301. I will put this on the teams radar for next sprint

**Comment by dangtony98** [comment]
Hey @Salman2301!

I made some changes to the code. Can you help fix a few edge-cases:

1. When **Scope > Repository Environment** is selected, then the **Environment** dropdown should be disabled until the **Repository** field is selected.
2. The **Repository Environment** field should never display organizations in the dropdown (for some reason it shows me organizations when **Repository** is not selected.

Screenshot illustrating 1:

Screenshot 2024-03-15 at 6 50 30 PM
f8b8de-7a60-46e0-bd89-eba7c3a7390a">

**Comment by Salman2301** [comment]
> Hey @Salman2301!
> 
> I made some changes to the code. Can you help fix a few edge-cases:
> 
> 1. When **Scope > Repository Environment** is selected, then the **Environment** dropdown should be disabled until the **Repository** field is selected.
> 2. The **Repository Environment** field should never display organizations in the dropdown (for some reason it shows me organizations when **Repository** is not selected.
> 

Both issue fixed with a simple disabled passed to select component.

**Pull request merged**